### ISSUE TITLE
bootstrap: internal compiler error: '(*RegArgs).Dump': nil register for value: v15 = VarDef <mem> {.autotmp_4} v12

### ISSUE THREAD

**Issue by mdartmann** [golang/go#50255]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.16.5 gccgo (GCC) 11.2.0 linux/amd64
</pre>

### Does this issue reproduce with the latest release?
yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/mae/.cache/go-build"
GOENV="/home/mae/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/mae/.go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/mae/.go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/libexec/gcc/x86_64-pc-linux-gnu/11.2.0"
GOVCS=""
GOVERSION="go1.16.5 gccgo (GCC) 11.2.0"
GCCGO="/usr/bin/gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/dev/null"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build3125549311=/tmp/go-build -gno-record-gcc-switches -funwind-tables"
</pre></details>

### What did you do?

I tried compiling go 1.17.5 and 1.18beta1 with the gccgo version mentioned above, using this script (`$1` points to an install directory that gets put into a tarball):
```sh
export GOARCH=amd64
export GOAMD64=v1
export GOROOT_FINAL=/usr/lib/go
export GOROOT_BOOTSTRAP=/usr
export GOPATH="$PWD"
export GOROOT="$PWD/go"

cd src
./make.bash  --no-clean -v

install -d "$1/usr/bin" "$1/usr/lib/go" "$1/usr/share/doc/go" \
    "$1/usr/lib/go/pkg/linux_amd64_"{dynlink,race}

cp -a bin pkg src lib misc api test "$1/usr/lib/go"
cp -r doc/* "$1/usr/share/doc/go"

ln -sf /usr/lib/go/bin/go "$1/usr/bin/go"
ln -sf /usr/lib/go/bin/gofmt "$1/usr/bin/gofmt"
ln -sf /usr/share/doc/go "$1/usr/lib/go/doc"

install -Dm644 VERSION "$1/usr/lib/go/VERSION"
```

### What did you expect to see?

I expected to compile a working version of go.

### What did you see instead?

I got the following error on `internal/abi.go`:
```log

/home/mae/.cache/kiss/proc/6866/build/go/src/internal/abi/abi.go:49:14: internal compiler error: '(*RegArgs).Dump': nil register for value: v15 = VarDef <mem> {.autotmp_4} v12
(*RegArgs).Dump method(*RegArgs) func()
  b1:
    (-47) v5 = ArgIntReg <*RegArgs> {r+0} [0] : AX (r[*RegArgs])
    (-47) v142 = StoreReg <*RegArgs> v5 : r[*RegArgs]
    (?) v1 = InitMem <mem>
    (48) v6 = CALLstatic <mem> {AuxCall{runtime.printlock}} v1
    (48) v7 = SelectN <mem> [0] v6
    (?) v117 = SB <uintptr> : SB
    (-48) v26 = LEAQ <*uint8> {go.string."Ints:"} v117 : AX
    (-48) v143 = MOVQconst <int> [5] : BX
    (48) v9 = CALLstatic <mem> {AuxCall{runtime.printstring}} [16] v26 v143 v7 : <>
    (48) v10 = SelectN <mem> [0] v9
    (48) v11 = CALLstatic <mem> {AuxCall{runtime.printunlock}} v10
    (48) v12 = SelectN <mem> [0] v11
    (-49) v15 = VarDef <mem> {.autotmp_4} v12
    (?) v2 = SP <uintptr> : SP
    (+49) v56 = MOVQload <uint64> v26 v15 : AX
    (49) v8 = MOVQstore <mem> {.autotmp_4} v26 v15 v15
    (49) v165 = MOVOload <int128> [8] v26 v8 : X0
    (49) v155 = MOVOstore <mem> {.autotmp_4} [8] v26 v8 v8
    (49) v163 = MOVOload <int128> [24] v26 v155 : X0
    (49) v146 = MOVOstore <mem> {.autotmp_4} [24] v26 v155 v155
    (49) v164 = MOVOload <int128> [40] v26 v146 : X0
    (49) v152 = MOVOstore <mem> {.autotmp_4} [40] v26 v146 v146
    (49) v162 = MOVOload <int128> [56] v26 v152 : X0
    (49) v17 = MOVOstore <mem> {.autotmp_4} [56] v26 v152 v152
    (?) v141 = MOVQconst <int> [0] : AX
    Plain -> b2
  b4: <- b2
    (49) v27 = MOVQloadidx8 <uintptr> {.autotmp_4} v26 v152 v132 : CX (x[uintptr])
    (+50) v28 = CALLstatic <mem> {AuxCall{runtime.printlock}} v132
    (50) v29 = SelectN <mem> [0] v28
    (50) v30 = CALLstatic <mem> {AuxCall{runtime.printsp}} v29
    (50) v31 = SelectN <mem> [0] v30
    (50) v33 = CALLstatic <mem> {AuxCall{runtime.printuint}} [8] v26 v31 : <>
    (50) v34 = SelectN <mem> [0] v33
    (50) v35 = CALLstatic <mem> {AuxCall{runtime.printunlock}} v34
    (50) v36 = SelectN <mem> [0] v35
    (+49) v39 = ADDQconst <int> [1] v26 : AX
    Plain -> b2
  b2: <- b1 b4
    (-47) v132 = Phi <mem> v17 v36
    (-49) v20 = Phi <int> v141 v39 : AX
    (+49) v24 = CMPQconst <flags> [9] v26
    LT v24 -> b4 b5 (likely)
  b5: <- b2
    (-47) v41 = VarKill <mem> {.autotmp_4} v132
    (+52) v42 = CALLstatic <mem> {AuxCall{runtime.printlock}} v41
    (52) v43 = SelectN <mem> [0] v42
    (52) v44 = CALLstatic <mem> {AuxCall{runtime.printnl}} v43
    (52) v45 = SelectN <mem> [0] v44
    (52) v46 = CALLstatic <mem> {AuxCall{runtime.printunlock}} v45
    (52) v47 = SelectN <mem> [0] v46
    (+53) v48 = CALLstatic <mem> {AuxCall{runtime.printlock}} v47
    (53) v49 = SelectN <mem> [0] v48
    (53) v51 = CALLstatic <mem> {AuxCall{runtime.printstring}} [16] v26 v31 v49 : <>
    (53) v52 = SelectN <mem> [0] v51
    (53) v53 = CALLstatic <mem> {AuxCall{runtime.printunlock}} v52
    (53) v54 = SelectN <mem> [0] v53
    (-54) v58 = VarDef <mem> {.autotmp_5} v54
    (+54) v131 = MOVQload <uint64> [72] v26 v58 : AX
    (+54) v129 = MOVQstore <mem> {.autotmp_5} v26 v58 v58
    (+54) v55 = ADDQconst <*[15]uint64> [80] v26 : SI
    (+54) v60 = DUFFCOPY <mem> [112] v26 v55 v129
    (?) v40 = MOVQconst <int> [0] : AX
    Plain -> b6
  b8: <- b6
    (54) v69 = MOVQloadidx8 <uint64> {.autotmp_5} v26 v55 v130 : CX (x[uint64])
    (+55) v70 = CALLstatic <mem> {AuxCall{runtime.printlock}} v130
    (55) v71 = SelectN <mem> [0] v70
    (55) v72 = CALLstatic <mem> {AuxCall{runtime.printsp}} v71
    (55) v73 = SelectN <mem> [0] v72
    (55) v74 = CALLstatic <mem> {AuxCall{runtime.printuint}} [8] v26 v73 : <>
    (55) v75 = SelectN <mem> [0] v74
    (55) v76 = CALLstatic <mem> {AuxCall{runtime.printunlock}} v75
    (55) v77 = SelectN <mem> [0] v76
    (+54) v79 = ADDQconst <int> [1] v26 : AX
    Plain -> b6
  b6: <- b5 b8
    (-47) v130 = Phi <mem> v60 v77
    (-54) v62 = Phi <int> v40 v79 : AX
    (+54) v4 = CMPQconst <flags> [15] v26
    LT v4 -> b8 b9 (likely)
  b9: <- b6
    (-47) v81 = VarKill <mem> {.autotmp_5} v130
    (+57) v82 = CALLstatic <mem> {AuxCall{runtime.printlock}} v81
    (57) v83 = SelectN <mem> [0] v82
    (57) v84 = CALLstatic <mem> {AuxCall{runtime.printnl}} v83
    (57) v85 = SelectN <mem> [0] v84
    (57) v86 = CALLstatic <mem> {AuxCall{runtime.printunlock}} v85
    (57) v87 = SelectN <mem> [0] v86
    (+58) v88 = CALLstatic <mem> {AuxCall{runtime.printlock}} v87
    (58) v89 = SelectN <mem> [0] v88
    (58) v91 = CALLstatic <mem> {AuxCall{runtime.printstring}} [16] v26 v73 v89 : <>
    (58) v92 = SelectN <mem> [0] v91
    (58) v93 = CALLstatic <mem> {AuxCall{runtime.printunlock}} v92
    (58) v94 = SelectN <mem> [0] v93
    (-59) v98 = VarDef <mem> {.autotmp_6} v94
    (+59) v138 = MOVQload <uint64> [192] v26 v98 : AX
    (+59) v137 = MOVQstore <mem> {.autotmp_6} v26 v98 v98
    (+59) v169 = MOVOload <int128> [200] v26 v137 : X0
    (+59) v161 = MOVOstore <mem> {.autotmp_6} [8] v26 v137 v137
    (+59) v167 = MOVOload <int128> [216] v26 v161 : X0
    (+59) v149 = MOVOstore <mem> {.autotmp_6} [24] v26 v161 v161
    (+59) v168 = MOVOload <int128> [232] v26 v149 : X0
    (+59) v158 = MOVOstore <mem> {.autotmp_6} [40] v26 v149 v149
    (+59) v166 = MOVOload <int128> [248] v26 v158 : X0
    (+59) v100 = MOVOstore <mem> {.autotmp_6} [56] v26 v158 v158
    (?) v106 = MOVQconst <int> [0] : AX
    Plain -> b10
  b12: <- b10
    (59) v108 = MOVQloadidx8 <unsafe.Pointer> {.autotmp_6} v26 v158 v128 : CX (x[unsafe.Pointer])
    (+60) v109 = CALLstatic <mem> {AuxCall{runtime.printlock}} v128
    (60) v110 = SelectN <mem> [0] v109
    (60) v111 = CALLstatic <mem> {AuxCall{runtime.printsp}} v110
    (60) v112 = SelectN <mem> [0] v111
    (60) v113 = CALLstatic <mem> {AuxCall{runtime.printpointer}} [8] v26 v112 : <>
    (60) v114 = SelectN <mem> [0] v113
    (60) v115 = CALLstatic <mem> {AuxCall{runtime.printunlock}} v114
    (60) v116 = SelectN <mem> [0] v115
    (+59) v118 = ADDQconst <int> [1] v26 : AX
    Plain -> b10
  b10: <- b9 b12
    (-47) v128 = Phi <mem> v100 v116
    (-59) v101 = Phi <int> v106 v118 : AX
    (+59) v140 = CMPQconst <flags> [9] v26
    LT v140 -> b12 b13 (likely)
  b13: <- b10
    (-47) v120 = VarKill <mem> {.autotmp_6} v128
    (+62) v121 = CALLstatic <mem> {AuxCall{runtime.printlock}} v120
    (62) v122 = SelectN <mem> [0] v121
    (62) v123 = CALLstatic <mem> {AuxCall{runtime.printnl}} v122
    (62) v124 = SelectN <mem> [0] v123
    (62) v125 = CALLstatic <mem> {AuxCall{runtime.printunlock}} v124
    (62) v126 = SelectN <mem> [0] v125
    (+63) v127 = MakeResult <mem> v126
    Ret v127
name r[*RegArgs]: [v5]
name x[uintptr]: [v27]
name x[uint64]: [v69]
name x[unsafe.Pointer]: [v108]



Please file a bug report including a short program that triggers the error.
https://golang.org/issue/new

go tool dist: FAILED: /home/mae/.cache/kiss/proc/6866/build/go/pkg/tool/linux_amd64/compile -std -pack -o /tmp/go-tool-dist-176103521/internal/abi/_go_.a -p internal/abi -importcfg /tmp/go-tool-dist-176103521/internal/abi/importcfg /home/mae/.cache/kiss/proc/6866/build/go/src/internal/abi/abi.go /home/mae/.cache/kiss/proc/6866/build/go/src/internal/abi/abi_amd64.go: exit status 2
/home/mae/.cache/kiss/proc/6866/build/go/src/runtime/internal/atomic/types.go:341:1: internal compiler error: '(*Float64).Store': resultInArg0 value's input SP cannot be an output of v16 = XCHGQ <uint64,mem> v14 v6 v12

Please file a bug report including a short program that triggers the error.
https://golang.org/issue/new

go tool dist: FAILED: /home/mae/.cache/kiss/proc/6866/build/go/pkg/tool/linux_amd64/compile -std -pack -o /tmp/go-tool-dist-176103521/runtime/internal/atomic/_go_.a -p runtime/internal/atomic -importcfg /tmp/go-tool-dist-176103521/runtime/internal/atomic/importcfg -asmhdr /tmp/go-tool-dist-176103521/runtime/internal/atomic/go_asm.h -symabis /tmp/go-tool-dist-176103521/runtime/internal/atomic/symabis /home/mae/.cache/kiss/proc/6866/build/go/src/runtime/internal/atomic/atomic_amd64.go /home/mae/.cache/kiss/proc/6866/build/go/src/runtime/internal/atomic/doc.go /home/mae/.cache/kiss/proc/6866/build/go/src/runtime/internal/atomic/stubs.go /home/mae/.cache/kiss/proc/6866/build/go/src/runtime/internal/atomic/types.go /home/mae/.cache/kiss/proc/6866/build/go/src/runtime/internal/atomic/types_64bit.go /home/mae/.cache/kiss/proc/6866/build/go/src/runtime/internal/atomic/unaligned.go: exit status 2
/home/mae/.cache/kiss/proc/6866/build/go/src/internal/cpu/cpu_x86.go:48:24: internal compiler error: 'doinit': panic during regalloc while compiling doinit:

interface conversion: ssa.Location is nil, not *	bootstrap_1cmd_1compile_1internal_1ssa	ssa.Register

goroutine 1 [running]:
bootstrap/cmd/compile/internal/ssa.Compile..func1
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/internal/ssa/compile.go:46
panic
	:0
bootstrap_1cmd_1compile_1internal_1ssa.regAllocState.regalloc
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/internal/ssa/regalloc.go:1581
bootstrap/cmd/compile/internal/ssa.regalloc
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/internal/ssa/regalloc.go:148
bootstrap_1cmd_1compile_1internal_1ssa.Compile
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/internal/ssa/compile.go:98
bootstrap/cmd/compile/internal/ssagen.buildssa
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/internal/ssagen/ssa.go:602
bootstrap_1cmd_1compile_1internal_1ssagen.Compile
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/internal/ssagen/pgen.go:183
bootstrap/cmd/compile/internal/gc.compileFunctions..func6
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/internal/gc/compile.go:153
bootstrap/cmd/compile/internal/gc.compileFunctions..func2
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/internal/gc/compile.go:125
bootstrap/cmd/compile/internal/gc.compileFunctions..func5
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/internal/gc/compile.go:152
bootstrap/cmd/compile/internal/gc.compileFunctions
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/internal/gc/compile.go:163
bootstrap_1cmd_1compile_1internal_1gc.Main
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/internal/gc/main.go:315
main.main
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/main.go:55



Please file a bug report including a short program that triggers the error.
https://golang.org/issue/new

go tool dist: FAILED: /home/mae/.cache/kiss/proc/6866/build/go/pkg/tool/linux_amd64/compile -std -pack -o /tmp/go-tool-dist-176103521/internal/cpu/_go_.a -p internal/cpu -importcfg /tmp/go-tool-dist-176103521/internal/cpu/importcfg -asmhdr /tmp/go-tool-dist-176103521/internal/cpu/go_asm.h -symabis /tmp/go-tool-dist-176103521/internal/cpu/symabis /home/mae/.cache/kiss/proc/6866/build/go/src/internal/cpu/cpu.go /home/mae/.cache/kiss/proc/6866/build/go/src/internal/cpu/cpu_x86.go: exit status 2
/home/mae/.cache/kiss/proc/6866/build/go/src/runtime/internal/math/math.go:14:33: internal compiler error: 'MulUintptr': panic during regalloc while compiling MulUintptr:

interface conversion: ssa.Location is nil, not *	bootstrap_1cmd_1compile_1internal_1ssa	ssa.Register

goroutine 1 [running]:
bootstrap/cmd/compile/internal/ssa.Compile..func1
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/internal/ssa/compile.go:46
panic
	:0
bootstrap_1cmd_1compile_1internal_1ssa.regAllocState.regalloc
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/internal/ssa/regalloc.go:1581
bootstrap/cmd/compile/internal/ssa.regalloc
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/internal/ssa/regalloc.go:148
bootstrap_1cmd_1compile_1internal_1ssa.Compile
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/internal/ssa/compile.go:98
bootstrap/cmd/compile/internal/ssagen.buildssa
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/internal/ssagen/ssa.go:602
bootstrap_1cmd_1compile_1internal_1ssagen.Compile
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/internal/ssagen/pgen.go:183
bootstrap/cmd/compile/internal/gc.compileFunctions..func6
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/internal/gc/compile.go:153
bootstrap/cmd/compile/internal/gc.compileFunctions..func2
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/internal/gc/compile.go:125
bootstrap/cmd/compile/internal/gc.compileFunctions..func5
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/internal/gc/compile.go:152
bootstrap/cmd/compile/internal/gc.compileFunctions
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/internal/gc/compile.go:163
bootstrap_1cmd_1compile_1internal_1gc.Main
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/internal/gc/main.go:315
main.main
	/home/mae/.cache/kiss/proc/6866/build/go/src/cmd/compile/main.go:55



Please file a bug report including a short program that triggers the error.
https://golang.org/issue/new

go tool dist: FAILED: /home/mae/.cache/kiss/proc/6866/build/go/pkg/tool/linux_amd64/compile -std -pack -o /tmp/go-tool-dist-176103521/runtime/internal/math/_go_.a -p runtime/internal/math -importcfg /tmp/go-tool-dist-176103521/runtime/internal/math/importcfg /home/mae/.cache/kiss/proc/6866/build/go/src/runtime/internal/math/math.go: exit status 2
```

Please let me know if this version of gccgo is unsupported or if you need some other debug output.

**Comment by ianlancetaylor** [comment]
This is a dup of #47771.  gccgo 11.2 had a bug that caused bootstraps to fail.  You need a version of gccgo newer than 11.2.  Sorry.

**Comment by mdartmann** [comment]
I'm sorry, but the newest version of gcc available is 11.2.0. See for yourself: http://ftp.gnu.org/gnu/gcc/
Do I need to bootstrap from a binary release?

**Comment by ianlancetaylor** [comment]
@mdartmann For the record I know that there is no version newer than 11.2 as yet.  Sorry, I should have mentioned that.  I'm not in charge of the GCC release schedule.  You can use an older release such as GCC 9.  Unfortunately GCC 10.4 has the same bug (and it will be fixed in GCC 10.5).
